### ISSUE TITLE
VirtualBox: does not work with Linux 5.4 kernel (on NixOS)

### ISSUE THREAD

**Issue by alexeymuranov** [NixOS/nixpkgs#97406]
**Describe the bug**
Enabling VirtualBox under Linux 5.4 kernel in NixOS causes boot-time problems, apparently related to network connection. There is a delay of 1m30s and a message:
```
A start job is running for
/dev/vboxnetctl
/sys/subsystem/net/devices/vboxnet0
```
Moreover, if Ethernet port is connected to a network that requires manual choice in NetworkManager, there is an extra error message just before the delay.

Afterwards virtual machines can be created but do not start, complaining about "Kernel driver not installed":
> Failed to open a session for the virtual machine Windows 10 (on NixOS).
> 
> The virtual machine 'Windows 10 (on NixOS)' has terminated unexpectedly during startup with exit code 1 (0x1).
> 
> 
> Result Code: 
> NS_ERROR_FAILURE (0x80004005)
> Component: 
> MachineWrap
> Interface: 
> IMachine {85632c68-b5bb-4316-a900-5eb28d3413df}

> Kernel driver not installed (rc=-1908)
> 
> The VirtualBox Linux kernel driver is either not loaded or not set up correctly. Please try setting it up again by executing
> 
> '/sbin/vboxconfig'
> 
> as root.
> 
> If your system has EFI Secure Boot enabled you may also need to sign the kernel modules (vboxdrv, vboxnetflt, vboxnetadp, vboxpci) before you can load them. Please see your Linux system's documentation for more information.
> 
> where: suplibOsInit what: 3 VERR_VM_DRIVER_NOT_INSTALLED (-1908) - The support driver is not installed. On linux, open returned ENOENT. 
> 

Everything seems to work fine under Linux 4.19 instead of 5.4

**To Reproduce**
1. Edit `/etc/nixos/configuration.nix`:
     ```nix
    { # ...
      boot.kernelPackages = pkgs.linuxPackages_5_4;
      virtualisation.virtualbox.host.enable = true;
    }
    ```

2. Rebuild and reboot. Observe boot-time issues.

3. Create a virtual machine in VirtualBox and try to start it. Observe the errors.

**Expected behavior**
Should work normally.

**Screenshots**
![Screenshot from 2020-09-07 23-34-43](https://user-images.githubusercontent.com/712698/92418184-7596d980-f166-11ea-82fa-0778770ce632.png)
![Screenshot from 2020-09-07 23-32-20](https://user-images.githubusercontent.com/712698/92418180-716abc00-f166-11ea-94d1-5e855c70034b.png)

**Additional context**
The following configuration works without issues:
```nix
{ # ...
  boot.kernelPackages = pkgs.linuxPackages_4_19;
  virtualisation.virtualbox.host.enable = true;
}
```

**Notify maintainers**
@aszlig 
@flokli 
@volth 

**Metadata**
The problem was confirmed on NixOS 20.03 and on NixOS 20.09:
```console
$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.4.62-hardened, NixOS, 20.03.2910.42674051d12 (Markhor)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.6`
 - channels(alexey): `""`
 - channels(root): `"nixos-20.03.2910.42674051d12"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```
```console
$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.4.62-hardened, NixOS, 20.09alpha47.c411fe8ae08 (Nightingale)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.7`
 - channels(alexey): `""`
 - channels(root): `"nixos-20.09alpha47.c411fe8ae08"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

**Comment by davidak** [comment]
I don't have those issues. Maybe it's related to the **hardened** kernel?

 - system: `"x86_64-linux"`
 - host os: `Linux 5.4.41, NixOS, 20.03.1950.48723f48ab9 (Markhor)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.4`
 - channels(root): `"home-manager-20.03, nixos-20.03.1950.48723f48ab9, nixos-hardware, nixos-unstable-20.09pre226148.0f5ce2fac0c"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

**Comment by alexeymuranov** [comment]
@davidak , it is written in the issue description that the problem is not specific to the hardened kernel.

**Comment by alexeymuranov** [comment]
@davidak, sorry, i removed the comment. In any case, I've just tested, and i have the same problem with:
```console
$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.4.64, NixOS, 20.09alpha235.e0508c81809 (Nightingale)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.7`
 - channels(alexey): `""`
 - channels(root): `"nixos-20.09alpha235.e0508c81809"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

**Comment by asymmetric** [comment]
I've just bumped into this

```
❯ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.4.87, NixOS, 20.09.20210123.a058d00 (Nightingale)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.4pre20201201_5a6ddb3`
 - channels(root): `"nixos-20.09.2538.0cfd08f4881, nixos-hardware"`
 - channels(asymmetric): `"home-manager-20.09, nixos-unstable-21.03pre262020.f211631c1cb"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
 - ```

**Comment by asymmetric** [comment]
The issue went away after rebooting :man_shrugging:

**Comment by alexeymuranov** [comment]
The issue seems resolved on NixOS 21.05.1590.63ee5cd99a2, Linux 5.4.132.
